lxd/containers.go

```diff
@@ -159,6 +292,123 @@ type containerStatePutReq struct {
 	Force   bool   `json:"force"`
 }
 
+type lxdContainer struct {
+	c *lxc.Container
+	id int
+}
+
+func (c *lxdContainer) Start() error {
+	return c.c.Start()
+}
+
+func (c *lxdContainer) Reboot() error {
+	return c.c.Reboot()
+}
+
+func (c *lxdContainer) Freeze() error {
+	return c.c.Freeze()
+}
+
+func (c *lxdContainer) Shutdown(timeout time.Duration) error {
+	return c.c.Shutdown(timeout)
+}
+
+func (c *lxdContainer) Stop() error {
+	return c.c.Stop()
+}
+
+func (c *lxdContainer) Unfreeze() error {
+	return c.Unfreeze()
+}
+
+func newLxdContainer(name string, daemon *Daemon) (*lxdContainer, error) {
+	d := &lxdContainer{}
+
+	certdbname := shared.VarPath("lxd.db")
+	db, err := sql.Open("sqlite3", certdbname)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+	rows, err := db.Query("select id, architecture from containers where name = ?", name);
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	d.id = -1
+	arch := 0
+	for rows.Next() {
+		var id int
+		rows.Scan(&id, &arch)
+		d.id = id
+	}
+	if d.id == -1 {
+		return nil, fmt.Errorf("Unknown container")
+	}
+
+	c, err := lxc.NewContainer(name, daemon.lxcpath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.SetConfigItem("lxc.include", "/usr/share/lxc/config/ubuntu.common.conf")
+	if err != nil {
+		return nil, err
+	}
+
+	var txtarch string
+	switch (arch) {
+	case 0:
+		txtarch = "x86_64"
+	default:
+		txtarch = "x86_64"
+	}
+	err = c.SetConfigItem("lxc.arch", txtarch)
+	if err != nil {
+		return nil, err
+	}
+
+	/* todo - get and translate containers_config entries
+	 * for now we hardcode some sane entries */
+	rootfsPath := shared.VarPath("lxc", name, "rootfs")
+	err = c.SetConfigItem("lxc.rootfs", rootfsPath)
+	if err != nil {
+		return nil, err
+	}
+	err = c.SetConfigItem("lxc.devttydir", "")
+	if err != nil {
+		return nil, err
+	}
+	uidstr := fmt.Sprintf("u 0 %d %d\n", daemon.id_map.Uidmin, daemon.id_map.Uidrange)
+	err = c.SetConfigItem("lxc.id_map", uidstr)
+	if err != nil {
+		return nil, err
+	}
+	gidstr := fmt.Sprintf("g 0 %d %d\n", daemon.id_map.Gidmin, daemon.id_map.Gidrange)
+	err = c.SetConfigItem("lxc.id_map", gidstr)
+	if err != nil {
+		return nil, err
+	}
+	err = c.SetConfigItem("lxc.loglevel", "0")
+	if err != nil {
+		return nil, err
+	}
+	logfile := shared.VarPath("lxc", name, "log")
+	err = c.SetConfigItem("lxc.logfile", logfile)
+	if err != nil {
+		return nil, err
+	}
+	err = c.SetConfigItem("lxc.utsname", name)
+	if err != nil {
+		return nil, err
+	}
+
+	d.c = c
+	c.SaveConfigFile("/tmp/config1")
```

Comment:
That could be a bit of a problem when creating multiple containers in parallel :)

Isn't there a mktemp equivalent in Go?